### ISSUE TITLE
Report error if TypeVars are used within bound or constrained type when defining a TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1208]
The following should be flagged as illegal:

```python
T1 = TypeVar("T1")
T2 = TypeVar("T2", bound=List[T1])
```

**Comment by erictraut** [comment]
Never mind, this already works.
